libvast/include/vast/detail/settings.hpp

```diff
@@ -27,4 +31,58 @@ void merge_settings(const caf::settings& src, caf::settings& dst,
 caf::expected<uint64_t>
 get_bytesize(caf::settings opts, std::string_view key, uint64_t defval);
 
+/// @brief Tries to extract a list value from config_value and convert it into
+/// std::vector<T>.
+/// @tparam T expected type of the list values.
+/// @param cfg_value value which should be a list type of Ts.
+/// @param error_message_on_list_extraction_failure error context of the error
+/// in case the cfg_value is not a list type.
+/// @return all list values extracted as Ts or error in 2 cases: the cfg_value
+/// is not of a list type, any value in the list is not of type T.
+template <class T>
+caf::expected<std::vector<T>> unpack_config_list_to_vector(
+  const caf::config_value& cfg_value,
+  std::string_view error_message_on_list_extraction_failure) {
+  const auto* list = caf::get_if<caf::config_value::list>(&cfg_value);
+  if (!list)
+    return caf::make_error(ec::invalid_configuration,
+                           error_message_on_list_extraction_failure.data());
+
+  std::vector<T> ret;
+  ret.reserve(list->size());
+  for (const auto& e : *list) {
+    const auto* val = caf::get_if<T>(&e);
+    if (!val)
+      return caf::make_error(ec::invalid_configuration,
+                             "Unexpected type in unpack_config_list_to_vector");
```

Comment:
This should rather say: "Type mismatch while unpacking config list: expected {caf::pretty_type_name<T>}, got {e}".